File: src/main/php7/binary_gap.php
#!/usr/bin/env php
<?php

function rem_norm(/*float*/ $value) /*: int*/  {
        return round($value/2 - floor($value /2),0);
    } 
    
class binary_gap {
    /**
    * Returns the length of the biggest gap of zeros between two ones
    */
    public function execute(/*float*/ $value) /*:  int*/ {
        
        while (rem_norm($value) > 0) {
            $value = $value / 2;        
        }
        $currentGap = 0;
        $maxGap = 0;
        while ($value > 0) {
            $remainder = rem_norm($value) ;
            if ($remainder == 0) {
                $currentGap++;
            } 
            else if ($currentGap != 0) {
                $maxGap = ($currentGap > $maxGap) ? $currentGap : $maxGap;
                $currentGap = 0;
            }
            $value = $value / 2;
        }
        return $maxGap;
    }
}

$value = 654345;
$app = new binary_gap;
echo $app->execute(floatval($value));

?>

File: sources/pl/binary_gap.pl
#!/usr/bin/env perl
use strict;

print &binary_gap(654345);

sub binary_gap { 
    my $value = shift;
       
    $value/=2 while ($value > 0 && $value % 2 == 0);
            
    my $currentGap = 0;
    my $maxGap = 0;
    while (int($value) > 0) {
        my $remainder = int($value % 2);
        if ($remainder == 0) {
            $currentGap++;
        } elsif ($currentGap != 0) {
            $maxGap = $maxGap < $currentGap ? $currentGap : $maxGap;
            $currentGap = 0;
        }
        $value/=2;
    }
    $maxGap;
}
